gravatar-ui/src/main/java/com/gravatar/ui/components/atomic/SocialMedia.kt

```
@@ -0,0 +1,117 @@
+    Fediverse("fediverse", R.drawable.fediverse_icon),
+    Foursquare("foursquare", R.drawable.foursquare_icon),
+    Github("github", R.drawable.github_icon),
+    Instagram("instagram", R.drawable.instagram_icon),
+    Mastodon("mastodon", R.drawable.mastodongeneric_icon),
+    StackOverflow("stackoverflow", R.drawable.stackoverflow_icon),
+    TikTok("tiktok", R.drawable.tiktok_icon),
+    TripIt("tripit", R.drawable.tripit_icon),
+    Tumblr("tumblr", R.drawable.tumblr_icon),
+    Twitch("twitch", R.drawable.twitch_icon),
+    Twitter("twitter", R.drawable.twitter_icon),
+    Vimeo("vimeo", R.drawable.vimeo_icon),
+    WordPress("wordpress", R.drawable.wordpress_icon),
+    ;
+
+    companion object {
+        public val shortnames = entries.associateBy { it.shortname }
+
+        public fun valueOf(shortname: String?): LocalIcon? {
+            return shortnames[shortname]
+        }
+    }
+}
+
+public class SocialMedia(val url: URL, val name: String, val iconUrl: URL? = null, val icon: LocalIcon? = null)
+
+public fun mediaList(profile: UserProfile): List<SocialMedia> {
+    val mediaList = mutableListOf<SocialMedia>()
+    // Force the Gravatar icon
+    profile.profileUrl?.let {
+        mediaList.add(SocialMedia(URL(it), LocalIcon.Gravatar.name, icon = LocalIcon.Gravatar))
+    }
+    // List and filter the other accounts from the profile, keep the same order coming from UserProfile.accounts list
+    profile.accounts?.let {
+        for (account in it) {
+            if (LocalIcon.valueOf(account.shortname) != null) {
+                // Add local icon if the shortname exists in our predefined list
+                mediaList.add(SocialMedia(URL(account.url), account.name, icon = LocalIcon.valueOf(account.shortname)))
+            } else {
+                // Add a "remote" icon (using the url coming from the endpoint response)
+                mediaList.add(SocialMedia(URL(account.url), account.name, iconUrl = URL(account.iconUrl)))
```

🛑 If `accout.iconUrl` is `null` the app will crash

```
Caused by: java.net.MalformedURLException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
                                                                                                    	at java.net.URL.<init>(URL.java:635)
                                                                                                    	at java.net.URL.<init>(URL.java:498)
                                                                                                    	at java.net.URL.<init>(URL.java:447)
                                                                                                    	at com.gravatar.ui.components.atomic.SocialMediaKt.mediaList(SocialMedia.kt:66)

```

---

Good catch, yes It will crash if the backend returns a malformed URL as well, eg. "example.com" (` if no protocol is specified, or an unknown protocol is found, or spec is null.`) so I guess the only thing to do here is to catch and ignore the exception

---

BTW, Same issue if the `account.url` is malformed or null

---

Fixed in 3af96216cdc157d8ec5d9316b89084c000e32bbc